aten/src/ATen/native/BlasKernel.cpp

```diff
@@ -301,7 +302,7 @@ static constexpr auto kF16RegistersPerIterationShift = kF16ElementsPerIterationS
 static constexpr auto kF16RegistersPerIteration = 1 << kF16RegistersPerIterationShift;
 static_assert(kF16RegistersPerIteration == kF16ElementsPerIteration / kF16ElementsPerRegister);
 
-static inline double reduce(float16x8_t x[kF16RegistersPerIteration]) {
+static inline float reduce(float16x8_t x[kF16RegistersPerIteration]) {
```

Comment:
Wasn't this done on purpose to preserve accuracy?

Reply:
I believe @malfet asked for it. perhaps I got confused though.

Reply:
Ok, in any case, the numerical error on macos arm seem relevant and needs investigation right?

Reply:
> Ok, in any case, the numerical error on macos arm seem relevant and needs investigation right?

yes, it's fixed. I accidentally replaced the float16 kernel with the bfloat16 kernel due to name changes